src/main/java/com/emc/pravega/perf/WriterWorker.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.emc.pravega.perf;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class WriterWorker extends Worker implements Callable<Void> {
+
+    WriterWorker(int sensorId, int eventsPerSec, int secondsToRun,
+                 boolean isRandomKey, int messageSize, Instant start,
+                 PerfStats stats, String streamName, long totalEvents) {
+
+        super(sensorId, eventsPerSec, secondsToRun,
+                isRandomKey, messageSize, start,
+                stats, streamName, totalEvents, 0);
+    }
+
+    public abstract CompletableFuture writeData(String key, String data);
+
+    public abstract void flush();
+
+    public abstract long eventCountIncrementAndGet();
+
+    public abstract long eventCountGet();
+
+    @Override
+    public Void call() throws InterruptedException, ExecutionException, IllegalStateException {
```

Comment:
Here and elsewhere: You don't need to declare runtime exception in the signature.

Reply:
do want me to remove ExecutionException?

Reply:
All three of these do not need to be enumerated here.

Reply:
since the call() invokes future.get , it requires  InterruptedException, ExecutionException to checked , so i have removed only IllegalStateException